components/registry-facade/pkg/blobserve/blobserve.go

```diff
@@ -126,18 +127,40 @@ func (reg *Server) serve(w http.ResponseWriter, req *http.Request) {
 		return
 	}
 
-	var fs http.FileSystem
-	fs = blob
-	if workdir != "" {
-		fs = prefixingFilesystem{Prefix: workdir, FS: fs}
-	}
-
 	log.WithField("path", req.URL.Path).Debug("handling blobserve")
 	pathPrefix := fmt.Sprintf("/%s:%s", repo, tag)
 	if req.URL.Path == pathPrefix {
 		req.URL.Path += "/"
 	}
 
+	// http.FileServer has a special case where ServeFile redirects any request where r.URL.Path
+	// ends in "/index.html" to the same path, without the final "index.html".
+	// We do not want this behaviour to make the _ide_index.html mechanism in ws-proxy work.
```

Comment:
`_ide_index.html mechanism` - we need to rename this mechanism, since it is based on query now :)